### PR TITLE
SL-1422: Handle rate limit errors in greenhouse. Created a specific error for Rate Limiting.

### PR DESCRIPTION
**Pull request by migueltaoliveira**

### CODE REVIEW

File: lib/greenhouse_io/api/client.rb [hunk 1]
```diff
@@ -99,37 +100,38 @@ def path_id(id = nil)
...
     end
 
     def get_from_harvest_api(url, options = {})
       response = get_response(url, {
-        :query => permitted_options(options),
-        :basic_auth => basic_auth
-      })
-
-      set_headers_info(response.headers)
+                                query: permitted_options(options),
```
Comment (tjmateus) [inline lib/greenhouse_io/api/client.rb:108]: Bad indentation

File: lib/greenhouse_io/api/client.rb [hunk 2]
```diff
@@ -99,37 +100,38 @@ def path_id(id = nil)
...
+      handle_response(response)
     end
 
     def post_to_harvest_api(url, body, headers)
       response = post_response(url, {
-        :body => JSON.dump(body),
-        :basic_auth => basic_auth,
-        :headers => headers
-      })
+                                 body: JSON.dump(body),
```
Comment (tjmateus) [inline lib/greenhouse_io/api/client.rb:117]: Ditto

File: lib/greenhouse_io/errors/error.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module GreenhouseIo
+  module Errors
+    class GreenhouseError < StandardError
```
[the remark this answers is not in the document]
Comment (migueltaoliveira) [reply lib/greenhouse_io/errors/error.rb:3]: Yes! Well spotted!

File: lib/greenhouse_io/api/client.rb (hunk 1 above)
Comment (migueltaoliveira) [reply lib/greenhouse_io/api/client.rb:108]: Not sure why my IDE did this. I'm using the same indentation as in the platform, but I will check this.

**Comment by migueltaoliveira** [comment]
Pull request closed until this is fixed: https://screenloop.atlassian.net/browse/SL-1430

**Pull request closed without merging**